### PR TITLE
v6.0.0-beta.0

### PR DESCRIPTION
**Pull request by alexfauquette**
### Release the packages

- [ ] Checkout the last version of the working branch
- [ ] Make sure you have the latest dependencies installed: `yarn`.
- [ ] Build the packages: `yarn release:build`.
- [ ] Release the versions on npm: `yarn release:publish` (you need your 2FA device).
- [ ] Push the newly created tag: `yarn release:tag`.
- [ ] If `@mui/x-codemod` has been released, set the pre-release package `tag` to `latest` with: `npm dist-tag add @mui/x-codemod@<released_version> latest`.

### Publish the documentation

The documentation must be updated on the `docs-vX` branch (`docs-v4` for `v4.X` releases, `docs-v5` for `v5.X` releases, ...)

- [ ] Push the working branch on the documentation release branch to deploy the documentation with the latest changes.

```sh
git push upstream next:docs-next -f
```

You can follow the deployment process [on the Netlify Dashboard](https://app.netlify.com/sites/material-ui-x/deploys?filter=docs-v5)
Once deployed, it will be accessible at https://material-ui-x.netlify.app/ for the `docs-v5` deployment.

### Publish GitHub release

- [ ] After documentation is deployed, publish a new release on [GitHub releases page](https://github.com/mui/mui-x/releases)

### CODE REVIEW

**Comment by DanailH** [comment]
Can we do `beta`? Looks like the only PRs that have `breaking change` labels are these too, one of which is already approved: https://github.com/mui/mui-x/pull/7579, https://github.com/mui/mui-x/pull/7594

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,132 @@
...
+- [DataGrid] Remove tag limit from `isAnyOf` operator input (#7592) @m4theushw
+- [DataGrid] Use v6 terminology (#7473) @DanailH
+- [DataGridPremium] Keep focus on first selected cell (#7482) @m4theushw
+- [l10n] Update Swedish (sv-SE) locale (#7585) @MaanTyringe
+
+### `@mui/x-date-pickers@v6.0.0-alpha.16` / `@mui/x-date-pickers-pro@v6.0.0-alpha.16`
+
+#### Breaking changes
+
+- The `showToolbar` prop has been moved to the `toolbar` component slot props:
```
Comment (flaviendelangle) [inline CHANGELOG.md:79]: - The new pickers have replaced the legacy one

  If you were using the new pickers with their temporary name, you just have to change your imports
  
  ```diff
  - import { Unstable_NextDatePicker as NextDatePicker } from '@mui/x-date-pickers/NextDatePicker';
  + import { DatePicker } from '@mui/x-date-pickers/DatePicker';

  - import { Unstable_DesktopNextDatePicker as DesktopNextDatePicker } from '@mui/x-date-pickers/DesktopNextDatePicker';
  + import { DesktopDatePicker } from '@mui/x-date-pickers/DesktopDatePicker';

  // Same for all the other pickers with an `Unstable_` prefix
  ```

  If you were still using the legacy picker (`DatePicker`, `DesktopDatePicker`, ...), please take a look at our [migration guide](https://next.mui.com/x/migration/migration-pickers-v5/#picker-components) for detailed explanations on how to start using the new ones.


- The fields components are no longer unstable

  ```diff
  - import { Unstable_DateField as DateField } from '@mui/x-date-pickers/DateField';
  + import { DateField } from '@mui/x-date-pickers/DateField';
  ```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,132 @@
...
+
+- The `page` and `pageSize` props and their respective event handlers `onPageChange` and `onPageSizeChange` were removed.
+Use `paginationModel` and `onPaginationModelChange` instead.
+
+  ```diff
+  -<DataGrid page={page} pageSize={pageSize} onPageChange={handlePageChange} onPageSizeChange={handlePageSizeChange} />
+  +<DataGrid paginationModel={{ page, pageSize }} onPaginationModelChange={handlePaginationModelChange} />
+  ```
+
+- The property `initialState.pagination.page` and `initialState.pagination.pageSize` were also removed.
```
Comment (MBilalShafi) [inline CHANGELOG.md:48]:
```suggestion
- The properties `initialState.pagination.page` and `initialState.pagination.pageSize` were also removed.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,155 @@
...
+
+### `@mui/x-codemod@v6.0.0-beta.0`
+
+#### Changes
+
+- [codemod] Remove `disableExtendRowFullWidth` prop (#7508) @MBilalShafi
+
+### Docs
+
+- [docs] Clean the field page (#7605) @flaviendelangle
```
Comment (LukasTy) [inline CHANGELOG.md:143]:
```suggestion
- [docs] Clean-up the `field components` page (#7605) @flaviendelangle
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,155 @@
...
+- [docs] List all pickers toolbar pages in api docs side menu (#7577) @LukasTy
+- [docs] Remove "Flex layout" docs section and demo (#7477) @cherniavskii
+- [docs] Rework the pickers Getting Started page (#7140) @flaviendelangle
```
Comment (LukasTy) [inline CHANGELOG.md:146]:
```suggestion
- [docs] Rework the pickers `Getting Started` page (#7140) @flaviendelangle
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,155 @@
...
+
+### Core
+
+- [core] Add missing need triage label on RFC @oliviertassinari
```
Comment (LukasTy) [inline CHANGELOG.md:150]:
```suggestion
- [core] Add missing `status: needs triage` label on RFC @oliviertassinari
```

File: packages/grid/x-data-grid-generator/package.json [hunk 6]
```diff
@@ -32,7 +32,7 @@
   "dependencies": {
     "@babel/runtime": "^7.20.7",
     "@mui/base": "^5.0.0-alpha.113",
-    "@mui/x-data-grid-premium": "6.0.0-alpha.15",
+    "@mui/x-data-grid-premium": "6.0.0-beta.1",
```
Comment (LukasTy) [inline packages/grid/x-data-grid-generator/package.json:35]:
```suggestion
    "@mui/x-data-grid-premium": "6.0.0-beta.0",
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,185 @@
...
+#### Changes
+
+- [codemod] Remove `disableExtendRowFullWidth` prop (#7508) @MBilalShafi
+
+### Docs
+
+- [docs] Clean the field page (#7605) @flaviendelangle
+- [docs] List all pickers toolbar pages in api docs side menu (#7577) @LukasTy
+- [docs] Remove "Flex layout" docs section and demo (#7477) @cherniavskii
+- [docs] Rework the pickers Getting Started page (#7140) @flaviendelangle
```
Comment (alexfauquette) [inline CHANGELOG.md:176]:
```suggestion
- [docs] Rework the pickers "Getting Started" page (#7140) @flaviendelangle
```

**Comment by alexfauquette** [comment]
@cherniavskii I added your two last PRs

File: CHANGELOG.md [hunk 8]
```diff
@@ -3,6 +3,203 @@
...
+  To catch errors that happen during rendering use the [error boundary](https://reactjs.org/docs/error-boundaries.html).
+  The `components.ErrorOverlay` slot was also removed.
+
+- The `componentError` event was removed.
+  Use the [error boundary](https://reactjs.org/docs/error-boundaries.html) to catch errors thrown during rendering.
+
+- The `apiRef.current.showError` method was removed.
+  The UI for errors is no longer handled by the grid.
+
+- The `GridErrorOverlay` component was removed.
```
Comment (MBilalShafi) [inline CHANGELOG.md:81]: A small order change to keep ErrorOverlay component related changes together?
```suggestion
- The `GridErrorOverlay` component was removed.

- The `componentError` event was removed.
  Use the [error boundary](https://reactjs.org/docs/error-boundaries.html) to catch errors thrown during rendering.

- The `apiRef.current.showError` method was removed.
  The UI for errors is no longer handled by the grid.
```

**Review by MBilalShafi — APPROVED** [review]

File: CHANGELOG.md [hunk 9]
```diff
@@ -3,6 +3,203 @@
...
+- The `rowsPerPageOptions` prop was renamed to `pageSizeOptions`.
+
+  ```diff
+  -<DataGrid rowsPerPageOptions={[10, 20, 50]} />
+  +<DataGrid pageSizeOptions={[10, 20, 50]} />
+  ```
+
+- The `error` and `onError` props were removed - the grid no longer catches errors during rendering.
+  To catch errors that happen during rendering use the [error boundary](https://reactjs.org/docs/error-boundaries.html).
+  The `components.ErrorOverlay` slot was also removed.
```
Comment (MBilalShafi) [inline CHANGELOG.md:73]: This looks like a separate point.
```suggestion

- The `components.ErrorOverlay` slot was also removed.
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -3,6 +3,204 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 6.0.0-beta.0
+
+_Jan 19, 2023_
+
+We'd like to offer a big thanks to the 10 contributors who made this release possible. Here are some highlights ✨:
```
Comment (joserodolfofreitas) [inline CHANGELOG.md:10]:
```suggestion
After a long period in alpha, we're glad to announce the first MUI X v6 beta! 
We encourage you to try out this version, packed with improvements, bug fixes, and a few highlighted features ✨:

**Data Grid**
- [Access to the API Object in the community version](https://next.mui.com/x/react-data-grid/api-object/)
- [Improved column menu](https://next.mui.com/x/react-data-grid/column-menu/)
- [Cell selection range](https://next.mui.com/x/react-data-grid/cell-selection/) (Premium)

**Date and Time pickers**
- [Fields: the new default input for pickers](https://next.mui.com/x/react-date-pickers/fields/).
- [Improved layout customization](https://next.mui.com/x/react-date-pickers/custom-layout/)
- Edit date ranges with drag and drop (Pro).

You can check the migration guides for the [Data Grid](https://next.mui.com/x/migration/migration-data-grid-v5/) and [Date Pickers](https://next.mui.com/x/migration/migration-pickers-v5/)



We'd like to offer a big thanks to the 10 contributors who made this release possible. 
```

**Review by joserodolfofreitas — COMMENTED** [review]
Can we add a small section to highlight some of the features achieved in alpha?

File: CHANGELOG.md [hunk 11]
```diff
@@ -3,6 +3,219 @@
...
+
+**Data Grid**
+- [Access to the API Object in the community version](https://next.mui.com/x/react-data-grid/api-object/)
+- [Improved column menu](https://next.mui.com/x/react-data-grid/column-menu/)
+- [Cell selection range](https://next.mui.com/x/react-data-grid/cell-selection/) (Premium)
+
+**Date and Time pickers**
+- [Fields: the new default input for pickers](https://next.mui.com/x/react-date-pickers/fields/).
+- [Improved layout customization](https://next.mui.com/x/react-date-pickers/custom-layout/)
+- Edit date ranges with drag and drop (Pro).
```
Comment (joserodolfofreitas) [inline CHANGELOG.md:21]:
```suggestion
- [Edit date ranges with drag and drop](https://next.mui.com/x/react-date-pickers/date-range-calendar/) (Pro).
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -3,6 +3,219 @@
...
+
+You can check the migration guides for the [Data Grid](https://next.mui.com/x/migration/migration-data-grid-v5/) and [Date Pickers](https://next.mui.com/x/migration/migration-pickers-v5/)
```
[the remark this answers is not in the document]
Comment (joserodolfofreitas) [reply CHANGELOG.md:23]:
```suggestion
You can check the migration guides for the [Data Grid](https://next.mui.com/x/migration/migration-data-grid-v5/) and [Date Pickers](https://next.mui.com/x/migration/migration-pickers-v5/) in our documentation.
```

**Pull request merged**